components/src/core/drawer/drawer-nav-item.tsx

```diff
@@ -85,7 +86,7 @@ export const DrawerNavItem: React.FC<DrawerNavItemProps> = (props) => {
                     <InfoListItem
                         dense
                         {...navItem}
-                        rightComponent={rightIcon}
+                        rightComponent={navItem.rightComponent || rightIcon}
```

Comment:
So the spread on line 113 actually doesn't do anything. The InfoListItem props are actually part of the NavItem (navItem). So those props are actually being spread on line 88. And then line 89 overrides the right component, which is why you only see the expander.

You'll want to do something like this to combine them:
![image](https://user-images.githubusercontent.com/29152776/97768871-ace98d00-1afc-11eb-8750-83cb740640a5.png)

Then you can simplify the type on line 21:
![image](https://user-images.githubusercontent.com/29152776/97768891-d2769680-1afc-11eb-96e9-b06f6fc57afa.png)
and get rid of the `...infoListItemProps` stuff because it will never be anything other than an empty object.

Reply:
![image](https://user-images.githubusercontent.com/29152776/97768933-24b7b780-1afd-11eb-82c6-d444fc276ed0.png)